src/Feldspar/Onnx/Operators.hs

```diff
@@ -259,10 +261,13 @@ onnxBatchNormalization attrs xs gamma beta mean var = ys
         epsilon = value $ P.realToFrac $ getAttr attrs aaFloat 1e-5 "epsilon"
 
 -- | Flatten a tensor to a matrix
-onnxFlatten :: Pushy vec => Attrs -> vec a -> Push DIM2 a
-onnxFlatten attrs vec = flatPush (P.fromIntegral $ getAttr attrs aaInt 1 "axis") $ toPush vec
+onnxFlatten :: (Pushy vec, Syntax a) => Attrs -> vec a -> Pull DIM2 a
+onnxFlatten attrs vec = toPull $ store $ flatPush (P.fromIntegral $ getAttr attrs aaInt 1 "axis") $ toPush vec
 
 -- | Flattening a Push vector to two dimensions
+--   We use Push vectors for flattening even though fusion is lost since flattening based on
```

Comment:
Sorry, this function is Push -> Push and behaves the expected way (?), I meant this comment should go on the caller which writes things to memory despite being a function on Pull vectors.